src/main/kotlin/biokotlin/kmer/AbstractKmerSet.kt

```diff
@@ -0,0 +1,102 @@
+package biokotlin.kmer
+
+import biokotlin.seq.NucSeq
+
+/**
+ * Represents some collection of kmers generated from nucleic acid sequence(s).
+ */
+abstract class AbstractKmerSet(val kmerSize: Int, val bothStrands: Boolean, val stepSize: Int, val keepMinOnly: Boolean) {
+    internal abstract var sequenceLength: Long
+    internal abstract var ambiguousKmers: Long
+
+    /** Returns the number of ambiguous kmers in the set. */
+    fun ambiguousKmers(): Long {return ambiguousKmers}
+
+    /** Returns the total length of sequence(s) used to generate this set. */
+    fun sequenceLength(): Long {return sequenceLength}
+
+    /** Returns the number of unique kmers in the set (not the count of total kmers). */
+    abstract fun setSize(): Long
+
+    /** Returns true if [kmer] is in the set. */
+    abstract fun contains(kmer: Kmer): Boolean
+
+    /**
+     * Returns the count of [kmer] in the set.
+     * If the set does not store counts, returns 1 if [kmer] is in the set.
+     */
+    abstract fun getCountOf(kmer: Kmer): Int
+
+    /** Adds the specified [kmer] to the set. */
+    abstract internal fun addKmerToSet(kmer: Long)
+
+    /** Returns true if set contains no kmers. */
+    abstract fun isEmpty(): Boolean
+
+    /**
+     * Given a NucSeq [seq], load all non-ambiguous kmers of length kmerSize into the set.
+     * Returns the number of kmers containing ambiguous bases.
+     */
+    protected fun seqToKmerMap(seq: NucSeq): Long {
+        var hashMask = 0L
+        (0 until kmerSize * 2).forEach { _ -> hashMask = (hashMask shl 1) or 1 }
+        var ambiguousKmers =0L
+
+        var goodBpRun=0
+        var previousHash = 0L
+
+        for(i in 0 until seq.size() step stepSize) {
+            previousHash = updateKmerHash(previousHash, seq[i].char, hashMask)
+            //If ambiguous base are encountered - then they are skipped but counted
+            if(previousHash == -1L) {goodBpRun=0; previousHash = 0L; ambiguousKmers++; continue}
+            goodBpRun++
+            if(goodBpRun<kmerSize) {
```

Comment:
I would suggest reformatting this if statement.  
`if(goodBpRun < kmerSize) {
    if( i >= (kmerSize - 1) * stepSize) ambiguousKmers++
    continue
}`